gammapy/modeling/tests/test_fit.py

```diff
@@ -31,7 +31,9 @@ class MyDataset(Dataset):
 
     def __init__(self, name="test"):
         self._name = name
-        self._models = Models([MyModel(x=1.99, y=2.99e3, z=3.99e-2)])
+        model = MyModel(x=1.99, y=2.99e3, z=3.99e-2)
+        model.name = name
```

Comment:
the dataset name is the same than the model name? Is it what you wish?

Reply:
yes. the point is only to make sure model names are different as this is a constraint imposed on `Datasets.models`